internal/data-importer/cmd/pg/import.go

```diff
@@ -0,0 +1,429 @@
+	// To resolve this, we clear the User Secret, prompting the PG operator to regenerate new credentials
+	// and synchronize with the DB.
+	if err := resetUserSecrets(ctx, k8sClient, dbName, namespace); err != nil {
+		return fmt.Errorf("failed to reset user secrets: %w", err)
+	}
+	log.Info().Msg("User credentials have been reset")
+	return nil
+}
+
+// returns: [baseBackupName, DBDirectory]
+func parseBackupPath(fullPath string) (string, string) {
+	// for consistency
+	fullPath = strings.TrimSuffix(fullPath, "/")
+	if !strings.HasPrefix(fullPath, "/") {
+		fullPath = "/" + fullPath
+	}
+	base := filepath.Base(fullPath)
+	fullPath = strings.TrimSuffix(fullPath, "backup/db/"+base)
+	return base, fullPath
+}
+
+func getRepoName(
+	ctx context.Context,
+	c client.Client,
+	dbName, namespace string,
+) (string, error) {
+	pg := &pgv2.PerconaPGCluster{}
+	if err := c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: dbName}, pg); err != nil {
+		return "", fmt.Errorf("failed to get PerconaPGCluster %s/%s: %w", namespace, dbName, err)
+	}
+	repoIdx := len(pg.Spec.Backups.PGBackRest.Repos) + 1
+	if repoIdx < 0 {
+		repoIdx = 0
+	}
+	return fmt.Sprintf("repo%d", repoIdx), nil
+}
+
+func pauseDBReconciliation(
+	ctx context.Context,
+	c client.Client,
+	name, namespace string) error {
+	db := &everestv1alpha1.DatabaseCluster{}
+	if err := c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: name}, db); err != nil {
+		return fmt.Errorf("failed to get database cluster %s/%s: %w", namespace, name, err)
+	}
+	annots := db.GetAnnotations()
+	if annots == nil {
+		annots = make(map[string]string)
+	}
+	annots[consts.PauseReconcileAnnotation] = consts.PauseReconcileAnnotationValueTrue
+	db.SetAnnotations(annots)
+	if err := c.Update(ctx, db); err != nil {
+		return fmt.Errorf("failed to pause reconciliation for database cluster %s/%s: %w", namespace, name, err)
+	}
+	return nil
+}
+
+func unpauseDBReconciliation(
+	ctx context.Context,
+	c client.Client,
+	name, namespace string) error {
+	db := &everestv1alpha1.DatabaseCluster{}
+	if err := c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: name}, db); err != nil {
+		return fmt.Errorf("failed to get database cluster %s/%s: %w", namespace, name, err)
+	}
+	annots := db.GetAnnotations()
+	if annots == nil {
+		annots = make(map[string]string)
+	}
+	delete(annots, consts.PauseReconcileAnnotation)
+	db.SetAnnotations(annots)
+	if err := c.Update(ctx, db); err != nil {
+		return fmt.Errorf("failed to unpause reconciliation for database cluster %s/%s: %w", namespace, name, err)
+	}
+	return nil
+}
+
+const pgBackrestSecretContentTpl = `
+[global]
+%s-s3-key=%s
+%s-s3-key-secret=%s
+`
+
+func preparePGBackrestSecret(
+	ctx context.Context,
+	c client.Client,
+	repoName string,
+	secretName string,
+	accessKeyId, secretAccessKey string,
+	dbName, namespace string,
+) error {
+	content := fmt.Sprintf(pgBackrestSecretContentTpl, repoName, accessKeyId, repoName, secretAccessKey)
```

Comment:
Will this work if there are backup schedules in-place? Aren't we going to clear the secrets for those storages?

Reply:
Ah, you're right. I assumed that if it creates a completely new Secret and specifies it under the backups configuration, PG operator will somehow merge configs from both the Secrets (one provided by Everest operator and the other by the Data Importer). But that didn't happen, it used only the one provided here and complained that it needs the keys for the scheduled backups too.

I've fixed it now -- it updates the same Secret to include the S3 keys from the data importer